egs/wsj/s5/steps/nnet3/chain/train.py

```diff
@@ -68,8 +68,8 @@ def get_args():
                         without the log-softmax layer for the outputs.  As
                         l2-norm of the log-softmax outputs can dominate the
                         objective function.""")
-    parser.add_argument("--chain.xent-regularize", type=float,
-                        dest='xent_regularize', default=0.0,
+    parser.add_argument("--chain.xent-regularize", type=str,
```

Comment:
If you want you could add an action to convert the string to a float array for you. See common_lib.StrToBoolAction and common_lib.NullstrToNoneAction.